verticapy/jupyter/extensions/chart_magic.py

```diff
@@ -192,30 +192,427 @@ def chart_magic(
+
+        %%chart -k hist
+        SELECT
+            pclass,
+            SUM(survived)
+        FROM titanic GROUP BY 1;
+
+    .. ipython:: python
+        :suppress:
+
+        %%chart -k hist
+        SELECT
+            pclass,
+            SUM(survived)
+        FROM titanic GROUP BY 1;
+
+    .. ipython:: python
+        :suppress:
+
+        hist = _
+        hist.write_html("figures/jupyter_extensions_chart_magic_chart_magic_4.html")
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_chart_magic_chart_magic_4.html
+
+    .. code-block:: python
+
+        %%chart -k scatter
+        SELECT
+            PetalLengthCm,
+            PetalWidthCm,
+            Species
+        FROM iris;
+
+    .. ipython:: python
+        :suppress:
+
+        %%chart -k scatter
+        SELECT
+            PetalLengthCm,
+            PetalWidthCm,
+            Species
+        FROM iris;
+
+    .. ipython:: python
+        :suppress:
+
+        scatter = _
+        scatter.write_html("figures/jupyter_extensions_chart_magic_chart_magic_5.html")
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_chart_magic_chart_magic_5.html
+
+    .. code-block:: python
+
+        %%chart -k boxplot
+        SELECT * FROM titanic;
+
+    .. ipython:: python
+        :suppress:
+
+        %%chart -k boxplot
+        SELECT * FROM titanic;
+
+    .. ipython:: python
+        :suppress:
+
+        boxplot = _
+        boxplot.write_html("figures/jupyter_extensions_chart_magic_chart_magic_6.html")
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_chart_magic_chart_magic_6.html
+
+    Exporting to HTML
+    =================
+    Export a chart to HTML:
+
+    .. code-block:: python
+
+        %%chart -k scatter -o "my_graphic"
+        SELECT * FROM titanic;
+
+    .. ipython:: python
+        :suppress:
+
+        %%chart -k spearman
+        SELECT * FROM titanic;
+
+    .. ipython:: python
+        :suppress:
+
+        export = _
+        export.write_html("figures/jupyter_extensions_chart_magic_chart_magic_7.html")
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_chart_magic_chart_magic_7.html
+
+    .. code-block:: python
+
+        file = open("my_graphic.html", "r")
+        file.read()
+        file.close()
+
+    Using Variables
+    ===============
+    You can use variables in charts with the ':' operator:
+
+    .. ipython:: python
+
+        import verticapy.stats as st
+
+        class_fare = titanic.groupby("pclass",
+                                     [st.avg(titanic["fare"])._as("avg_fare")])
+
+    .. ipython:: python
+        :suppress:
+
+        html_file = open("figures/jupyter_extensions_chart_magic_chart_magic_8.html", "w")
+        html_file.write(class_fare._repr_html_())
+        html_file.close()
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_chart_magic_chart_magic_8.html
+
```

Comment:
You can then use the variable in the query.

.. note:: In this example, we use a vDataFrame, but it's still possible to use a pandas.DataFrame, a numpy.array, or many other in-memory objects.